### PR TITLE
Refactor single vault collateral types

### PR DESCRIPTION
**Pull request by brettvniekerk**
related to [this ticket](https://github.com/handle-fi/handle-react/issues/1107)

the network should not be an `Extract<Network, "ethereum">` but just a regular `Network` -- it should be the app's responsibility to handle errors / if the network is unsupported

### CODE REVIEW

**Review by Preston-Harrison — APPROVED** [review]

File: src/components/Vaults.ts [hunk 1]
```diff
@@ -334,10 +336,12 @@ export default class Vaults {
...
   public getSingleCollateralVaults = async (
     account: string,
-    network: SingleCollateralVaultNetwork,
+    network: Network,
     signer: ethers.Signer
-  ): Promise<SingleCollateralVault[]> => {
+  ): Promise<SingleCollateralVault[] | undefined> => {
     this.initialisationCheck();
+    if (network === "ethereum") return;
+
```
Comment (rellfy) [inline src/components/Vaults.ts:344]: Why the `if`?
Also, this should throw, not return undefined

File: src/components/Vaults.ts [hunk 2]
```diff
@@ -271,10 +271,12 @@ export default class Vaults {
...
     account: string,
     vaultSymbol: SingleCollateralVaultSymbol,
-    network: SingleCollateralVaultNetwork,
+    network: Network,
     signer: ethers.Signer
-  ): Promise<SingleCollateralVault> => {
+  ): Promise<SingleCollateralVault | undefined> => {
     this.initialisationCheck();
+    if (network === "ethereum") return;
+
```
Comment (rellfy) [inline src/components/Vaults.ts:279]: Also should throw, not return undefined

File: src/components/Vaults.ts (hunk 1 above)
Comment (brettvniekerk) [reply src/components/Vaults.ts]: refactored

File: src/components/Vaults.ts (hunk 2 above)
Comment (brettvniekerk) [reply src/components/Vaults.ts]: refactored

**Pull request closed without merging**